packages/ra-ui-materialui/src/field/TranslatableFieldsTabContent.tsx

```diff
@@ -41,34 +58,32 @@ export const TranslatableFieldsTabContent = (
         >
             {Children.map(children, field =>
                 field && isValidElement<any>(field) ? (
-                    <div key={field.props.source}>
-                        {addLabel ? (
-                            <Labeled
-                                resource={resource}
-                                label={field.props.label}
-                                source={field.props.source}
-                            >
-                                {cloneElement(field, {
-                                    ...field.props,
-                                    label: getLabel(field.props.source),
-                                    record,
-                                    source: getSource(
-                                        field.props.source,
-                                        locale
-                                    ),
-                                })}
-                            </Labeled>
-                        ) : typeof field === 'string' ? (
-                            field
-                        ) : (
-                            cloneElement(field, {
-                                ...field.props,
-                                label: getLabel(field.props.source),
-                                record,
-                                source: getSource(field.props.source, locale),
-                            })
-                        )}
-                    </div>
+                    <ResourceContextProvider
+                        value={resource}
+                        key={field.props.source}
+                    >
+                        <RecordContextProvider value={record}>
+                            <SourceContextProvider value={sourceContext}>
+                                <div>
+                                    {addLabel ? (
+                                        <Labeled
+                                            resource={resource}
+                                            label={field.props.source}
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** You changed `field.props.label‏` to `field.props.source` which I think is a bug because it introduces a regression in the default label key.

Following screenshots were taken in the `Basic` story, after removing the `i18nProvider`.

BEFORE

![2024-02-20_11-14](https://github.com/marmelab/react-admin/assets/14542336/6e980115-54ab-4229-8e4c-997d98f4dbad)


AFTER

![2024-02-20_11-14_1](https://github.com/marmelab/react-admin/assets/14542336/1b4888ae-e9de-45ee-9c38-294591586cd0)

Reply:
It's actually a bugfix as this is how all fields are handled elsewhere unless I missed something?